### PR TITLE
[AIRFLOW-6534] - BigQuery - move methods from BigQueryBaseCursor to BigQueryHook

### PR DESCRIPTION
**Pull request by TobKed**
---
Issue link: [AIRFLOW-6534](https://issues.apache.org/jira/browse/AIRFLOW-6534)

- [x] Description above provides context of the change
- [x] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/gcp/hooks/bigquery.py [hunk 1]
```diff
@@ -154,85 +160,8 @@ def table_exists(self, project_id: str, dataset_id: str, table_id: str) -> bool:
...
-        """ Return a new :py:class:`Cursor` object using the connection. """
-        return BigQueryCursor(*self._args, **self._kwargs)
-
-    def rollback(self) -> NoReturn:
-        """ BigQueryConnection does not have transactions """
-        raise NotImplementedError(
-            "BigQueryConnection does not have transactions")
-
-
-class BigQueryBaseCursor(LoggingMixin):
```
Comment (turbaszek) [inline airflow/gcp/hooks/bigquery.py:207]: Shouldn't we leave those classes to preserve backward compatibility?

Comment (TobKed) [reply airflow/gcp/hooks/bigquery.py:207]: I agree and they are kept (https://github.com/apache/airflow/pull/7131/files/1856949f169716d5a5a014d8e9d0a181e879a920#diff-8afbbf904c5f5075ffc575dc43dd5836R2065).
Apparently git is not handling moving big parts of code good enough hence reviewing this PR may be problematic. To make it easier I divided by making commits by moved methods.

**Comment by mik-laj** [comment]
Can you also move test methods to `TestBigQueryHookMethods` class?

**Comment by mik-laj** [comment]
Very good work!

**Comment by TobKed** [comment]
Thanks @mik-laj , I've moved tests and rebased on the latest master.

**Comment by mik-laj** [comment]
Ohhh...I have now seen that the note in the UPDATING.md file is missing. Can you add it?

**Comment by TobKed** [comment]
@mik-laj I've added information in `UPDATING.md`. Please verify is it correct.

File: UPDATING.md [hunk 2]
```diff
@@ -57,6 +57,11 @@ https://developers.google.com/style/inclusive-documentation
 
 -->
 
+### Move methods from BiqQueryBaseCursor to BigQueryHook
+
+To simplify BigQuery operators (no need of `Cursor`) and standardize usage of hooks within all GCP methods from `BiqQueryBaseCursor`
+were moved to `BigQueryHook`. Using them by from `Cursor` object is still possible due to preserve backward compatibility but will raise `DeprecationWarning`.
```
Comment (mik-laj) [inline UPDATING.md]: Can you add the list of methods?

Comment (TobKed) [reply UPDATING.md]: Sure. I rephrased information little bit and added table with old/new paths for moved methods.

**Review by mik-laj — APPROVED** [review]

**Comment by mik-laj** [comment]
Merge conflict :crying_cat_face:

**Pull request merged**